### DISCUSSION TITLE
Table Inside Drawer and problem with edge filter (first column)

### DISCUSSION THREAD

**Discussion by DrakMrak** [ant-design-blazor/ant-design-blazor#3861] [Q&A] [answered]
![FilterInsideDrawer](https://github.com/ant-design-blazor/ant-design-blazor/assets/1819596/6054cb45-3cd1-4bc1-b03e-be7a0b2ff680)
Hello
I have a problem with the filter size if the table is in the Drawer and the first column is at the edge of the drawer (the editing field is very small). Is there any way to solve this?  For example, by turning the filter to the other side for the outermost columns.
I can't seem to find a solution...

Thank you in advance for any information
Best Regards
DM

(C#, Blazor, AntDesign 0.19.0)

**Comment by DrakMrak** [comment]
So far I solved it by opening the Drawer on LEFT then the filters behave correctly. The problem on the right side persists. For me it was solved by a design change, however, it would probably need to be checked!

**Reply by ElderJames** [reply, marked as answer]
Thank you, I have open a PR for custom the filter input parameters https://github.com/ant-design-blazor/ant-design-blazor/pull/3897 which may can help.

**Comment by DrakMrak** [comment]
For me, the problem persisted 🤔 maybe the problem is the defined width of
the first column of the table (width="90")
Filter: Contain is very short
Filter: Equal is better (from number 105 I see 05)
see the pictures.

code:

<Drawer Closable="true" Width="550" Visible="dvisible2" Title='("Edit
stations list")' OnClose="_=>ClosePopup()" Placement="right">
    <Template style="height:90%">
        <Table @ref="tableSt" Context="selstationTable"
DataSource="stationsFiltered"
               TItem="Station"
               ScrollY="550"
               PageSize="15"
               HidePagination="false"
               ***@***.***"
               ***@***.***"
               @bind-SelectedRows="@_stationsSelected"
            <TitleTemplate>
                <Select TItem="StationGroup"
                        TItemValue="int"
                        ***@***.***"
                        @bind-Value="@_selectedSGValue"
                        ***@***.***(StationGroup.SysId)"
                        ***@***.***(StationGroup.GroupName)"
                        ***@***.***"

OnSelectedItemChanged="OnSelectedItemChangedHandlerSG"
                        Style="width: 300px;"
                        AllowClear
                        Placeholder="Choose station group">
                </Select>
                <hr style="margin-top: 6px; margin-bottom: 0px;" />
            </TitleTemplate>
            <ColumnDefinitions>
                <Selection Key="@(selstationTable.SysId.ToString())"
Type="checkbox" Disabled="@(selstationTable.IsCanceled)" />
                <PropertyColumn Title="No." Property="c=>c.DirectoryNum"
Filterable Sortable Width="90"></PropertyColumn>
                <PropertyColumn Title="Station Name"
Property="c=>c.DirectoryName" Ellipsis Filterable Sortable></PropertyColumn>

...........continue


út 4. 6. 2024 v 18:41 odesílatel James Yeung ***@***.***>
napsal:

> Sorry, I can't reproduce the issue
>
> image.png (view on web)
> <https://github.com/ant-design-blazor/ant-design-blazor/assets/7550366/ffc32221-46b9-4a43-a0b8-627f836b20f6>
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ant-design-blazor/ant-design-blazor/discussions/3861#discussioncomment-9667324>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAN4HTHRI7D5UZYAWIJGTKTZFXU2TAVCNFSM6AAAAABHVS7QTCVHI2DSMVQWIX3LMV43SRDJONRXK43TNFXW4Q3PNVWWK3TUHM4TMNRXGMZDI>
> .
> You are receiving this because you authored the thread.Message ID:
> <ant-design-blazor/ant-design-blazor/repo-discussions/3861/comments/9667324
> @github.com>
>


-- 
S pozdravem a přáním hezkého dne
Zdeněk Charvát

**Comment by DrakMrak** [comment]
![FilterInsideDrawerIII](https://github.com/ant-design-blazor/ant-design-blazor/assets/1819596/a27f60b5-ac26-4dfa-b293-8fa5bc5ac74f)
![FilterInsideDrawerII](https://github.com/ant-design-blazor/ant-design-blazor/assets/1819596/ba649a91-0c63-4cd7-8d94-db923c6d2cbd)
Three column table
---
First column I have limited width = 90
Second column is free
third column is limited width = 35 (show only some status icon)
---
```
<ColumnDefinitions>
    <Selection Key="@(selstationTable.SysId.ToString())" Type="checkbox" Disabled="@(selstationTable.IsCanceled)" />
    <PropertyColumn Title="No." Property="c=>c.DirectoryNum" Filterable Sortable Width="150"></PropertyColumn>
    <PropertyColumn Title="Station Name" Property="c=>c.DirectoryName" Ellipsis Filterable Sortable></PropertyColumn>
    <Column Title="C." TData="string" Width="35">
        @if (selstationTable.IsCanceled)
        {
            <Icon Type="close-square" Theme="twotone" TwotoneColor="#eb2f96" Height="1.4em" Width="1.4em" />
        }
    </Column>
</ColumnDefinitions>
```

**Comment by DrakMrak** [comment]
I tested width of columns
1) Width = 90 (box for number is small)
2) Width = 150 (OK)
so there's a dependency on the column width - I can customize the design to this but in my opinion the filter box if the drawer is on the left could slide out more into the open space at the moment of using the filter... 
Tested on version 0.19.1

--
![FiltrColumnWidth90](https://github.com/ant-design-blazor/ant-design-blazor/assets/1819596/ecd1ad4e-4de5-43c8-bb47-5fcce914a5b4)
![FiltrColumnWidth150](https://github.com/ant-design-blazor/ant-design-blazor/assets/1819596/72080ae4-47f8-4e60-9104-5cc31b26f135)
![code](https://github.com/ant-design-blazor/ant-design-blazor/assets/1819596/7882e50b-fb6e-4cd8-88df-10b9fc363f5d)

**Reply by ElderJames** [reply]
Thank you for the testing, I think we should give the input field a default width, or use the width set on column.

**Reply by DrakMrak** [reply]
This sounds reasonable as long as the column width is set. There may also be a problem with too large a column width. 🤔
